app/jobs/marc_export_job.rb

```diff
@@ -4,8 +4,8 @@ class MarcExportJob < ActiveJob::Base
   def perform(theses)
     marc_filename = "#{filename}.xml"
     zip_filename = "#{filename}.zip"
-    zip_data = MarcBatch.new(theses, marc_filename, zip_filename).build
-    BatchMailer.marc_batch_email(zip_filename, zip_data, theses).deliver_later
+    zip_file = MarcBatch.new(theses, marc_filename, zip_filename).build
+    BatchMailer.marc_batch_email(zip_filename, zip_file, theses).deliver_now
```

Comment:
I wonder if doing something like:
```ruby
begin
  zip_file = MarcBatch.new(theses, marc_filename, zip_filename).build
  BatchMailer.marc_batch_email(zip_filename, zip_file, theses).deliver_now
ensure
  zip_file.close
end
```

...rather than closing from the BatchMailer might both work and be easier to keep track of. Note: I have not tried this and there may be reasons it's not the way to go :)

Reply:
@JPrevost Yeah, I do like that better. My only hesitation is I think that would make the explicit close untestable, since `zip_file` isn't returned anywhere. Do you have any concerns about that?

Reply:
Not really. I feel like we can prove to ourselves this works by looking at our filesystem to ensure the file gets removed when we expect and then trust ruby to be consistent. Thanks for noting that concern though.